tempesta_fw/http_parser.c

```diff
@@ -2092,15 +2169,101 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len)
 				}
+				__FSM_MOVE_nofixup_n(Req_MethPurg, 4);
+			/*
+			 * Other popular methods: COPY, DELETE, LOCK, MKCOL,
+			 * MOVE, OPTIONS, PATCH, PROPFIND, PROPPATCH, PUT,
+			 * TRACE, UNLOCK.
+			 */
+			case TFW_CHAR4_INT('C', 'O', 'P', 'Y'):
+				req->method = TFW_HTTP_METH_COPY;
+				__FSM_MOVE_nofixup_n(Req_MUSpace, 4);
+			case TFW_CHAR4_INT('D', 'E', 'L', 'E'):
+				if (likely(__data_available(p, 6))
+				    && (*(p + 4) == 'T') && (*(p + 5) == 'E'))
+				{
+					req->method = TFW_HTTP_METH_DELETE;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 6);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethDele, 4);
+			case TFW_CHAR4_INT('L', 'O', 'C', 'K'):
+				req->method = TFW_HTTP_METH_LOCK;
+				__FSM_MOVE_nofixup_n(Req_MUSpace, 4);
+			case TFW_CHAR4_INT('M', 'K', 'C', 'O'):
+				if (likely(__data_available(p, 5))
+				    && (*(p + 4) == 'L'))
+				{
+					req->method = TFW_HTTP_METH_MKCOL;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 5);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethMkco, 4);
+			case TFW_CHAR4_INT('M', 'O', 'V', 'E'):
+				req->method = TFW_HTTP_METH_MOVE;
+				__FSM_MOVE_nofixup_n(Req_MUSpace, 4);
+			case TFW_CHAR4_INT('O', 'P', 'T', 'I'):
+				if (likely(__data_available(p, 8))
+				    && (*((unsigned int *)p + 1)
+					== TFW_CHAR4_INT('O', 'N', 'S', ' ')))
+				{
+					req->method = TFW_HTTP_METH_OPTIONS;
+					__FSM_MOVE_nofixup_n(Req_Uri, 8);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethOpti, 4);
+			case TFW_CHAR4_INT('P', 'A', 'T', 'C'):
+				if (likely(__data_available(p, 5))
+				    && (*(p + 4) == 'H'))
+				{
+					req->method = TFW_HTTP_METH_PATCH;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 5);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethPatc, 4);
+			case TFW_CHAR4_INT('P', 'R', 'O', 'P'):
+				if (likely(__data_available(p, 8))
+				    && (*((unsigned int *)p + 1)
+					== TFW_CHAR4_INT('F', 'I', 'N', 'D')))
+				{
+					req->method = TFW_HTTP_METH_PROPFIND;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 8);
+				}
+				if (likely(__data_available(p, 9))
+				    && (*((unsigned int *)p + 1)
+					== TFW_CHAR4_INT('P', 'A', 'T', 'C'))
+				    && (*(p + 8) == 'H'))
+				{
+					req->method = TFW_HTTP_METH_PROPPATCH;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 9);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethProp, 4);
+			case TFW_CHAR4_INT('P', 'U', 'T', ' '):
+				req->method = TFW_HTTP_METH_PUT;
+				__FSM_MOVE_nofixup_n(Req_Uri, 4);
+			case TFW_CHAR4_INT('T', 'R', 'A', 'C'):
+				if (likely(__data_available(p, 5))
+				    && (*(p + 4) == 'E'))
+				{
+					req->method = TFW_HTTP_METH_TRACE;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 5);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethTrac, 4);
+			case TFW_CHAR4_INT('U', 'N', 'L', 'O'):
+				if (likely(__data_available(p, 6))
+				    && (*(p + 4) == 'C') && (*(p + 5) == 'K'))
+				{
+					req->method = TFW_HTTP_METH_UNLOCK;
+					__FSM_MOVE_nofixup_n(Req_MUSpace, 6);
+				}
+				__FSM_MOVE_nofixup_n(Req_MethUnlo, 4);
 			}
-			TFW_PARSER_BLOCK(Req_Method); /* Unsupported method */
+			/* TODO: unknown */
```

Comment:
What's the sense of the TODO comment? We're falling to unknown method and validate it's alphabet - this seems the right logic, what we need to do more?